lib/mosaic-geomath/src/main/java/org/eclipse/mosaic/lib/spatial/PointCloud.java

```diff
@@ -0,0 +1,317 @@
+    }
+
+    /**
+     * Relative coordinates mean the use of a cartesian coordinate system with its origin (0,0,0) at
+     * the {@link PointCloud}'s reference, returned by {@link #getReference()}. In addition, the
+     * coordinate system is rotated. See {@link #getRotation()} for details.
+     *
+     * @return end points of all rays of this {@link PointCloud} in relative coordinates
+     * @see #getReference() origin of ray / translation of coordinates
+     * @see #getRotation() rotation of coordinates
+     */
+    public abstract List<Point> getRelativeEndPoints();
+
+
+    /**
+     * @return end points of all rays of this {@link PointCloud} that hit something in relative coordinates
+     */
+    public abstract List<Point> getRelativeEndPointsWithHit();
+
+    /**
+     * @return end points of all rays of this {@link PointCloud} in absolute coordinates
+     */
+    public abstract List<Point> getAbsoluteEndPoints();
+
+    /**
+     * @return end points of all rays of this {@link PointCloud} that hit something in absolute coordinates
+     */
+    public abstract List<Point> getAbsoluteEndPointsWithHit();
+
+    /**
+     * A {@link PointCloud} is represented by a constant origin and their end points
+     * returned by {@link #getRelativeEndPoints()} and {@link #getAbsoluteEndPoints()}
+     * Relative coordinates are translated and rotated absolute coordinates.
+     *
+     * @return origin of the rays forming this point cloud in absolute coordinates
+     * @see #getRotation()
+     */
+    public Vector3d getReference() {
+        return reference;
+    }
+
+    /**
+     * Ray end points in relative coordinates of this point cloud are translated and rotated end points in
+     * absolute coordinates.
+     * <p>
+     * Let o be the ray origin in absolute coordinates, let R be the rotation matrix. Thus, the transformation
+     * of relative coordinates r into absolute coordinates a is
+     * <code>a = R * r + o</code>
+     *
+     * @return rotation matrix R
+     * @see #getReference() returns <code>o</code>
+     */
+    public RotationMatrix getRotation() {
+        return rotation;
+    }
+
+    /**
+     * @return simulation time of {@link PointCloud}
+     */
+    public long getTimestamp() {
+        return timestamp;
+    }
+
+    /**
+     * @return sensor's minimum detectable range in distance units of Phabmacs
+     */
+    public double getMinRange() {
+        return minRange;
+    }
+
+    /**
+     * @return sensor's maximum detectable range in distance units of Phabmacs
+     */
+    public double getMaxRange() {
+        return maxRange;
+    }
+
+    /**
+     * A {@link Point} of the point cloud consists of its coordinates, an identifier
+     * of the type of object the point has hit, and the distance to the point cloud origin.
+     */
+    public static class Point extends Vector3d {
+
+        private static final long serialVersionUID = 1L;
+
+        private final byte hitType;
```

Comment:
How are the `hitTypes` defined? Shouldn't there be an enum?

Reply:
For storage we used `byte` to keep memory consumption low. Enum is way more expensive here. There currently is a enum in PHABMACS called "WorldObjectType", but moving this as well here would definitely break the "generalistic" argument for placing the point cloud in this package.